puppet/modules/jenkins_job_builder/manifests/config.pp

```diff
@@ -6,46 +6,28 @@
   String $password,
   Integer[0] $jenkins_jobs_update_timeout = 600,
   String $command_arguments = 'foreman-infra-jenkins-job-update',
+  String $git_project_name = 'foreman-infra',
+  String $git_repo = 'https://github.com/theforeman/foreman-infra.git',
+  Optional[String] $git_args = undef,
+  String $git_relative_path = 'ci',
 ) {
   $config_name = $name
   $directory = '/etc/jenkins_jobs'
   $inifile = "${directory}/jenkins_jobs_${config_name}.ini"
 
-  file { "${directory}/${config_name}":
-    ensure  => directory,
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0755',
-    recurse => true,
-    purge   => true,
-    force   => true,
-    source  => "puppet:///modules/jenkins_job_builder/${config_name}",
+  git::repo { "jenkins-jobs-${config_name}":
```

Comment:
What's the mechanism to keep this repo updated? If you don't a notify can easily revert the job config back to an old state.

Reply:
If I don't notify what?

Reply:
I suppose this is the way the git repo defines works (https://github.com/theforeman/puppet-git/blob/master/manifests/repo.pp) whereby it won't reclone on each run. How is this normally handled? Do I blow away the repository on each run to enforce the clone?

Reply:
Our module was really designed to be more of a "server" repo: it should be there so you can git push to it (via SSH). We never had to deal with pulling. You can use https://github.com/puppetlabs/puppetlabs-vcsrepo#git with ensure latest.